pkg/controller/permissions/permissions_check.go

```diff
@@ -44,105 +43,103 @@ func CopyPromoteStagedAppImage(req router.Request, resp router.Response) error {
 //
 //	b.1) granted by the user (as set in the app spec)
 //	b.2) authorized by the image role authorizations (if enabled)
-func CheckPermissions(transport http.RoundTripper) router.HandlerFunc {
-	return func(req router.Request, _ router.Response) error {
-		app := req.Object.(*v1.AppInstance)
-
-		iraEnabled, err := config.GetFeature(req.Ctx, req.Client, profiles.FeatureImageRoleAuthorizations)
-		if err != nil {
-			return err
-		}
-		if !iraEnabled {
-			app.Status.Staged.ImagePermissionsDenied = nil
-		}
+func CheckPermissions(req router.Request, _ router.Response) error {
```

Comment:
I think we may have kept the transport here and passed it on to checkImageAllowed and also to getAuthorizedPermissions, since it's useful as a remoteOpts for the signature checks.
Just seeing that you dropped it reminded me, that I didn't use it here :roll_eyes:

Reply:
Please add it back if you think it is necessary. Since things were working and we weren't using it, I removed it.